src/vos/evtree.c

```diff
@@ -1823,6 +1843,70 @@ evt_select_node(struct evt_context *tcx, const struct evt_rect *rect,
 	return rc < 0 ? nd1 : nd2;
 }
 
+/** Expand the dynamic tree root node if it is currently full and not
+ *  already at full size
+ */
+static inline int
+evt_node_extend(struct evt_context *tcx, struct evt_trace *trace)
+{
+	struct evt_node *nd_cur;
+	int              rc;
+	uint8_t          old_order;
+	void            *new_node;
+	int              old_size;
+	umem_off_t       new_off;
+
+	if (tcx->tc_order == tcx->tc_max_order)
+		return 0;
+
+	nd_cur = evt_off2node(tcx, trace->tr_node);
+	D_ASSERT(evt_node_is_leaf(tcx, nd_cur));
+	D_ASSERT(tcx->tc_depth == 1);
+	D_ASSERT(tcx->tc_feats & EVT_FEAT_DYNAMIC_ROOT);
+
+	if (!evt_node_is_full(tcx, nd_cur))
+		return 0;
+
+	old_order = tcx->tc_order;
+	old_size  = evt_node_size(tcx, true);
+
+	if (tcx->tc_order == 7)
```

Comment:
Why 7 is special?

Reply:
I could potentially add some defines.  Nothing special about 7 per-se, it's just the maximum of the dynamic values until I switch to user-specified order.